pkg/sql/sqlliveness/slstorage/key_encoder.go

```diff
@@ -37,21 +37,21 @@ type rbrEncoder struct {
 	rbrIndex roachpb.Key
 }
 
-func (e *rbrEncoder) encode(session sqlliveness.SessionID) (roachpb.Key, error) {
+func (e *rbrEncoder) encode(session sqlliveness.SessionID) (roachpb.Key, []byte, error) {
```

Comment:
encode should not return the []byte slice representing the region. The region bytes are pointing at memory that is owned by the string and since []byte is mutable it would be easy to accidentally violate the go memory model.

Since you want the value as a string I think it is possible to avoid allocation here by making a safe version of DecodeSessionID that extracts a substring from the session ID. The encode function could use the UnsafeBytes helper to avoid allocating memory for the region.